src/StorageUtility.js

```diff
@@ -1,11 +1,195 @@
 class StorageUtility {
```

Comment:
The algorithms going on here are complicated enough that I think it's work you walking me through this!! Maybe we can do that during our spring meeting and include Steph as well?

Reply:
Makes sense to me! I'd be more than happy to walk through it.